pkg/kubelet/status/manager.go

```diff
@@ -382,6 +382,11 @@ func (m *manager) syncPod(uid types.UID, status versionedPodStatus) {
 				glog.V(3).Infof("Pod %q is terminated, but some containers are still running", format.Pod(pod))
 				return
 			}
+			if pod.DeletionGracePeriodSeconds != nil && *pod.DeletionGracePeriodSeconds == 0 {
```

Comment:
See #21175. If after #21175, the flake in test-cmd.sh doesn't occur anymore, then it means the race we are trying to deal here is very unlikely to occur.